conftest.py

```diff
@@ -227,6 +235,21 @@ def sr_disk(request, host):
             f"disk or block device {disk} is either not present or already used on master host"
     yield disk
 
+@pytest.fixture(scope='session')
+def sr_disk_4k(request, host):
+    disk = request.param
+    if disk == "auto":
+        logging.info(">> Check for the presence of a free 4KiB block device on the master host")
+        disks = host.available_disks(4096)
+        assert len(disks) > 0, "a free 4KiB block device is required on the master host"
+        disk = disks[0]
+        logging.info(f">> Found free block device(s) on hostA1: {' '.join(disks)}. Using {disk}.")
+    else:
+        logging.info(f">> Check that block device {disk} is available on the master host")
+        assert disk in host.available_disks(4096), \
+            f"block device {disk} must be available for use on master host"
```

Comment:
The message should likely mention we're looking for a 4k disk here, and the pre-existing message shoud likewise mention 512B

Reply:
I changed the sr_disk_4k messages but this PR hasn't touched normal sr_disk.

Reply:
> this PR hasn't touched normal sr_disk

Sure, but my point was that as the context changes and we support both 512 and 4096, we may want to make it visible when a test is for 512.  OTOH since the standard code path will also apply to 4K-able SMAPIv3, changing the other message would indeed be wrong.

Reply:
Well, the new available_disks by default gives only 512 disks as to not break existing SMAPIv1 tests on machines with multiple devices with different blocksizes. Maybe something specific to SMAPIv3 needs to be done for SR disks. Or tests specifically aimed at 4KiB can be written using the sr_disk_4k fixture for SMAPIv3 drivers.

Reply:
So my comment indeed holds :)

Maybe what we indeed need is to have `None` be the default block size to mean "any", so size-agnostic tests are possible, and make sure to pass an explicit `512` for old SMAPIv1 drivers?

Reply:
That would work. We could leave the default being 512 for existing users of sr_disk but add a any parameter to available_disks.